client/ayon_deadline/lib.py

```diff
@@ -43,3 +48,368 @@ def get_instance_job_envs(instance) -> "dict[str, str]":
+    ScheduledStartDateTime: Optional[str] = field(
+        default=None)  # <dd/MM/yyyy HH:mm>
+    ScheduledDays: Optional[int] = field(default=1)  # Default: 1
+    JobDelay: Optional[str] = field(default=None)  # <dd:hh:mm:ss>
+    Scheduled: Optional[str] = field(
+        default=None)  # <Day of the Week><Start/Stop>Time=<HH:mm:ss>
+
+    # Scripts
+    PreJobScript: Optional[str] = field(default=None)  # Default: blank
+    PostJobScript: Optional[str] = field(default=None)  # Default: blank
+    PreTaskScript: Optional[str] = field(default=None)  # Default: blank
+    PostTaskScript: Optional[str] = field(default=None)  # Default: blank
+
+    # Event Opt-Ins
+    EventOptIns: Optional[str] = field(
+        default=None)  # Default blank (comma-separated list)
+
+    # Environment
+    EnvironmentKeyValue: Any = field(
+        default_factory=partial(DeadlineKeyValueVar, "EnvironmentKeyValue"))
+    IncludeEnvironment: Optional[bool] = field(default=False)  # Default: false
+    UseJobEnvironmentOnly: Optional[bool] = field(default=False)  # Default: false
+    CustomPluginDirectory: Optional[str] = field(default=None)  # Default blank
+
+    # Job Extra Info
+    ExtraInfo: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "ExtraInfo"))
+    ExtraInfoKeyValue: Any = field(
+        default_factory=partial(DeadlineKeyValueVar, "ExtraInfoKeyValue"))
+
+    OverrideTaskExtraInfoNames: Optional[bool] = field(
+        default=False)  # Default false
+
+    TaskExtraInfoName: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "TaskExtraInfoName"))
+
+    OutputFilename: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputFilename"))
+    OutputFilenameTile: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputFilename{}Tile"))
+    OutputDirectory: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputDirectory"))
+
+    AssetDependency: str = field(
+        default_factory=partial(DeadlineIndexedVar, "AssetDependency"))
+
+    TileJob: bool = field(default=False)
+    TileJobFrame: int = field(default=0)
+    TileJobTilesInX: int = field(default=0)
+    TileJobTilesInY: int = field(default=0)
+    TileJobTileCount: int = field(default=0)
+
+    MaintenanceJob: bool = field(default=False)
+    MaintenanceJobStartFrame: int = field(default=0)
+    MaintenanceJobEndFrame: int = field(default=0)
+
+
+@dataclass
+class AYONDeadlineJobInfo(DeadlineJobInfo):
+    """Contains additional AYON variables from Settings for internal logic."""
+
+    # AYON custom fields used for Settings
+    UsePublished: Optional[bool] = field(default=None)
+    UseAssetDependencies: Optional[bool] = field(default=None)
+    UseWorkfileDependency: Optional[bool] = field(default=None)
+
+    MachineList: Optional[str] = field(
+        default=None)  # Default blank (comma-separated list)
+    MachineListDeny: Optional[bool] = field(default=None)
```

Comment:
This seems odd - why is this "AYONDeadlineJobInfo"? This just regular Deadline job info?

Reply:
Not really, you use them to fill actuall DLJI(Deadline JobInfo). There is nothing as 'MachineList' in DLJI, but based on list of machine names AND checkbox if that list is DENY, you prepare value for `Whitelist` or `Blacklist` which is actual fields in DLJI.
We could argue, if we should have it as two separate fields(lists), it is mimicking how it looks in Maya Submitter right now.
![image](https://github.com/user-attachments/assets/6524d9f8-ac27-4e1f-a843-4f8fe07f0d31)

Reply:
Ah I see. That makes more sense. What's keeping us from correcting these values directly instead of having to first store the 'wrong' values on job info?